forest/cli/src/cli/info_cmd.rs

```diff
@@ -0,0 +1,406 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use chrono::{DateTime, Utc};
+use clap::Subcommand;
+use colored::*;
+use forest_blocks::tipset_json::TipsetJson;
+use forest_cli_shared::cli::CliOpts;
+use forest_rpc_client::{
+    chain_get_name, chain_head, node_ops::node_status, start_time, wallet_balance,
+    wallet_default_address,
+};
+use forest_shim::econ::TokenAmount;
+use forest_utils::misc::LoggingColor;
+
+use fvm_shared::clock::EPOCH_DURATION_SECONDS;
+use fvm_shared::{clock::ChainEpoch, BLOCKS_PER_EPOCH};
+use num::BigInt;
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+
+use super::Config;
+use crate::cli::handle_rpc_err;
+use crate::humantoken::TokenAmountPretty;
+
+#[derive(Debug, Subcommand)]
+pub enum InfoCommand {
+    Show,
+}
+
+#[derive(Debug)]
+pub struct NodeStatusInfo {
+    /// duration in seconds of how far behind the node is with respect to
+    /// syncing to head
+    pub behind: Duration,
+    /// Chain health is the percentage denoting how close we are to having
+    /// an average of 5 blocks per tipset in the last couple of
+    /// hours. The number of blocks per tipset is non-deterministic
+    /// but averaging at 5 is considered healthy.
+    pub health: f64,
+    /// epoch the node is currently at
+    pub epoch: ChainEpoch,
+    /// base fee
+    pub base_fee: TokenAmount,
+    /// sync status information
+    pub sync_status: SyncStatus,
+    /// Start time of the node
+    pub start_time: DateTime<Utc>,
+    /// Current network the node is running on
+    pub network: String,
+    /// Default wallet address selected.
+    pub default_wallet_address: Option<String>,
+    /// Default wallet address balance
+    pub default_wallet_address_balance: Option<String>,
+}
+
+pub struct NodeInfoOutput {
```

Comment:
this seems a bit bloaty - can you think of any ways to slim it down?

Reply:
I've noticed https://docs.rs/colored/latest/colored/control/struct.ShouldColorize.html - should we be integrating this into our code instead of presumably writing this a bunch of times?

Reply:
Not sure about this. I wanted to have this for keeping individual formatting of NodeStatusInfo fields separate and also to aid in writing formatting tests against.

Reply:
> I wanted to have this for keeping individual formatting of NodeStatusInfo fields separate and also to aid in writing formatting tests against.

It's good to make sure that your code is testable, but if you're writing colored-string aware tests, that might be the sign that the test approach is wrong.

Think of other aspects of UI - do we write progress bar tests? How about testing log output? These are all out of scope for our codebase - we rely on libraries for these. Test your data layer, but (save for user-reported bugs), presentation layer tests aren't really _interesting_

Our tests should cover our logic, and basic implementation details.

I think we should move the colored printing to `run`, keep the `NodeStatusInfo::new` tests, and remove presentation layer tests

Reply:
In addition to integration tests, we can and should write unit tests for this code. We have an algorithm for computing the network health and a pretty-printer for formatting the status of a node. Both functions are pure and should be relatively easy to test.

I don't see how `NodeInfoOutput` relates to testing, though. As I said in a previous review, the code would be smaller and easier to test without that struture.

Reply:
Removed NodeInfoOutput, fixed presentation layer unit tests and added integration tests, but the integration tests fails (https://github.com/ChainSafe/forest/actions/runs/5196426787/jobs/9370116138#step:8:1585), because it needs the daemon to be running. Not sure if we still want to have integration test for this part of forest-cli.